app/components/affiliate-link-page/accept-referral-container.hbs

```diff
@@ -24,32 +24,7 @@
     </div>
 
     <div class="inline-flex mt-8">
-      <PrimaryButtonWithSpinner
-        @size="regular"
-        @shouldShowSpinner={{this.isCreatingAffiliateReferral}}
-        @isDisabled={{not this.acceptOfferButtonIsEnabled}}
-        class="px-5 py-3"
-        data-test-accept-referral-button
-        {{on "click" this.handleAcceptOfferButtonClick}}
-      >
-        {{#if this.currentUserIsAnonymous}}
-          {{svg-jar "github" class="fill-current w-5 transform transition-all mr-2"}}
-        {{/if}}
-
-        Accept offer →
-      </PrimaryButtonWithSpinner>
-
-      {{#if this.currentUserIsAnonymous}}
-        <EmberTooltip @text="Click to login via GitHub" />
-      {{else if this.currentUserIsReferrer}}
-        <EmberTooltip @text="You can't accept your own referral offer." />
-      {{else if this.currentUserIsAlreadyEligibleForReferralDiscount}}
-        <EmberTooltip
-          @text="You've already accepted {{this.currentUser.currentAffiliateReferral.affiliateLink.usernameForDisplay}}'s referral offer."
-        />
-      {{else if this.currentUserCanAccessMembershipBenefits}}
-        <EmberTooltip @text="As a CodeCrafters member, you already have full access." />
-      {{/if}}
+      <AffiliateLinkPage::AcceptReferralButton @affiliateLink={{@affiliateLink}} data-test-accept-referral-button />
```

Comment:
```suggestion
      <AffiliateLinkPage::AcceptReferralButton @affiliateLink={{@affiliateLink}} />
```

`data-test-accept-referral-button` could go inside the `AcceptReferralButton` component so that it automatically applies everywhere, no? Is there a reason to have the caller pass this in?